### PR TITLE
JBPM-8001 - Extend Repository service to allow more operations and pl…

### PR DESCRIPTION
**Pull request by tsurdilo**
…uggable storage

### CODE REVIEW

**Comment by tsurdilo** [comment]
depends on https://github.com/kiegroup/jbpm/pull/1395

**Comment by tsurdilo** [comment]
Jenkins retest please

**Comment by tsurdilo** [comment]
Jenkins retest please

**Comment by tsurdilo** [comment]
Jenkins retest please

**Comment by tsurdilo** [comment]
we had this before..even tho jbpm pr branch is named the same it seems its not being used....

**Comment by tsurdilo** [comment]
@mswiderski made updates per our conversations:
* no longer using module and version -- using category
* serviceinfo js is now included in the produced repository jar (had to make some changes to poms in each module to get that)

we should be good now. This should go in first as will need to rebase the okta workitem pr after

File: repository/src/main/java/org/jbpm/process/workitem/repository/service/RepoData.java [hunk 1]
```diff
@@ -19,21 +19,35 @@
 
 public class RepoData {
 
+    private String id;
     private String name;
     private String displayName;
     private String defaultHandler;
     private String documentation;
+    private String module;
+    private String version;
```
Comment (mswiderski) [inline repository/src/main/java/org/jbpm/process/workitem/repository/service/RepoData.java]: do we still need these two? module and version?

**Pull request merged**